contents/stable_marriage_problem/code/ruby/stable_marriage.rb

```diff
@@ -0,0 +1,113 @@
+class Generator
+    def initialize(size)
+        @size = size
+    end
+
+    def generate(prefix, r)
+        Array.new(@size){|i|
+            Person.new(
+                i,
+                "#{prefix} #{i}",
+                (0 ... @size).to_a.shuffle!(random: r)
+            )
+        }
+    end
+end
+
+class Person
+    def initialize(id, name, prefs)
+        @id      = id
+        @name    = name
+        @prefs   = prefs
+        @partner = LONELY
+        @choices = 0
+    end
+
+    def lonely
+        @partner == LONELY
+    end
+
+    def propose(partners)
+        if !self.lonely
+            raise '%s is not lonely!' % self.name
+        end
+        choice = @prefs[@choices]
+        partners[choice].onPropose(self)
+        @choices += 1
+    end
+
+    # Represents this Person as an Array of strings
+    def print
+        name = if self.lonely
+            then "Lonely"
+            else @partner.name
+        end
+        [
+            "##{@id} | #{@name}",
+            "Partner: #{name}"
+        ]
+    end
+
+    protected
+    attr_reader :id, :name
+    attr_writer :partner
+
+    # Acts upon a given Proposal
+    def onPropose(partner)
+        if !self.lonely
```

Comment:
```suggestion
        unless lonely
```

(plus or minus a question mark, if you take the previous suggestion about that)